### PR TITLE
Functional test - Add new test 'CRUD virtual product' in product page V2

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'CRUD virtual product' in product page V2
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no need
| Possible impacts? | no

### CODE REVIEW

File: tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import ProductFaker from '@data/faker/product';
+
+import type {Page} from 'playwright';
+
+/**
+ * Virtual product tab on new product V2 page, contains functions that can be used on the page
+ * @class
+ * @extends BOBasePage
+ */
+export default class VirtualProductTab extends BOBasePage {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts]:
```suggestion
class VirtualProductTab extends BOBasePage {
```
No need to export class unless you extending it somewhere

File: tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      await this.waitForSelectorAndClick(page, this.productChooseFile);
+      await this.waitForVisibleSelector(page, this.productFile);
+      await this.uploadFile(page, this.productFile, productData.fileName);
+      await this.setValue(page, this.productFileNameInput, productData.fileName);
+      await this.setValue(page, this.productFileDownloadTimesLimit, productData.allowedDownload);
+    }
+  }
+}
+
+module.exports = new VirtualProductTab();
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts]:
```suggestion
export default new VirtualProductTab();
```

**Review by Progi1984 — COMMENTED** [review]
The feedback of @boubkerbribri is perfect.

File: tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts]: Done thx :)

File: tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts]: Done :)

**Comment by Progi1984** [comment]
@nesrineabdmouleh ESLint is not happy :angry:

File: tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+// Import page
+import BOBasePage from '@pages/BO/BObasePage';
+import ProductFaker from '@data/faker/product';
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/productsV2/add/virtualProductTab.ts]:
```suggestion

// Import data
import ProductFaker from '@data/faker/product';
```

File: tests/UI/campaigns/productV2/functional/05_CRUDVirtualProduct.ts [hunk 4]
```diff
@@ -0,0 +1,429 @@
+import type {BrowserContext, Page} from 'playwright';
+import {expect} from 'chai';
+
+// Import utils
+import helper from '@utils/helpers';
+import basicHelper from '@utils/basicHelper';
+import files from '@utils/files';
+
+// Import test context
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/05_CRUDVirtualProduct.ts:9]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/05_CRUDVirtualProduct.ts [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+import loginCommon from '@commonTests/BO/loginBO';
+import {enableNewProductPageTest, disableNewProductPageTest} from '@commonTests/BO/advancedParameters/newFeatures';
+
+// Import faker data
+import ProductFaker from '@data/faker/product';
+
+// Import demo data
+import {Statuses} from '@data/demo/orderStatuses';
+import {DefaultCustomer} from '@data/demo/customer';
+import {PaymentMethods} from '@data/demo/paymentMethods';
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/05_CRUDVirtualProduct.ts]:
```suggestion
// Import data
import {Statuses} from '@data/demo/orderStatuses';
import {DefaultCustomer} from '@data/demo/customer';
import {PaymentMethods} from '@data/demo/paymentMethods';
import ProductFaker from '@data/faker/product';
```

**Review by Progi1984 — COMMENTED** [review]
Small feedbacks

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**